### PR TITLE
[MDS-5609] Minespace Loading 🚄🏍🏎️💨🏁⚡

### PR DESCRIPTION
**Pull request by simensma-fresh**
## Objective 

[MDS-5609](https://bcmines.atlassian.net/browse/MDS-5609)

*What?*
Bunch of configuration and app tweaks to improve loading speed of minespace.
- Got rid of the redirect that happens to check if a user is logged in
- Lazy load PDF Viewer and leaflet only when needed
- Applied code splitting based on different routes
- Various import updates (lodash + antd icons, moment) to make sure they were tree shaked properly
- Deploy source maps as separate files
- Make sure webpack removes comments from built files
- Make css non-render blocking
- Bunch of different tweaks to make GTMetrix happy

Bundle analyzer
- Added build step to analyze the built bundle and upload a stats.json that can be analyzed with https://chrisbateman.github.io/webpack-visualizer/

*Before*
image

*After*
image

TODO:
- [x] A bunch of cleanup
- [ ] Apply these changes to core
- [x] Test out serving the frontend from S3 + Cloudfront instead of Openshift for things like improved caching and `http/2`

### CODE REVIEW

**Review by henryoforeh-dev — APPROVED** [review]

File: services/minespace-web/src/components/common/Loading.js [hunk 1]
```diff
@@ -1,20 +1,14 @@
...
-  const defaultOptions = {
-    loop: true,
-    autoplay: true,
-    animationData: loader,
-  };
   return (
     <div id="loading-screen">
       <div id="loader">
-        <Lottie options={defaultOptions} />
+        <img src={LOADER} alt="Loading" />
```
Comment (taraepp) [inline services/minespace-web/src/components/common/Loading.js:11]: 🙌 
I will not miss the Lottie-related console errors!

**Review by taraepp — APPROVED** [review]
🚀

File: services/minespace-web/src/constants/routes.js [hunk 2]
```diff
@@ -1,19 +1,34 @@
...
-import MinesPage from "@/components/pages/MinesPage";
-import ProjectPage from "@/components/pages/Project/ProjectPage";
-import ProjectSummaryPage from "@/components/pages/Project/ProjectSummaryPage";
-import ReturnPage from "@/components/pages/ReturnPage";
-import TailingsSummaryPageWrapper from "@/components/pages/Tailings/TailingsSummaryPageWrapper";
-import IncidentPage from "@/components/pages/Incidents/IncidentPage";
-import IncidentSuccessPage from "@/components/pages/Incidents/IncidentSuccessPage";
-import { UsersPage } from "@/components/pages/UsersPage";
+const DamsPage = React.lazy(() => import("@common/components/tailings/dam/DamsPage"));
+const InformationRequirementsTablePage = React.lazy(() =>
```
Comment (matbusby-fw) [inline services/minespace-web/src/constants/routes.js:4]: Yay lazy loading!

**Review by matbusby-fw — APPROVED** [review]
🚀

**Pull request merged**